File: GenerateAesKey/GenerateRandomAesKey.php
<?php
function generateRandomAesKey()
{
    return openssl_random_pseudo_bytes(32, $crypto_strong);
}

function base64Encoding($input)
{
    return base64_encode($input);
}

echo 'Generate a 32 byte long AES key' . PHP_EOL;
$aesKey = generateRandomAesKey();
$aesKeyBase64 = base64Encoding($aesKey);
echo 'generated key length: ' . strlen($aesKey) . ' data: '. $aesKeyBase64 . PHP_EOL;
?>

File: GenerateAesKey/GenerateRandomAesKey.cs
using System;
using System.Security.Cryptography;

public class GenerateAesKey {
	public static void Main() {
		Console.WriteLine("Generate a 32 byte long AES key");
    byte[] aesKey = GenerateRandomAesKey();
    string aesKeyBase64 = Base64Encoding(aesKey);
    Console.WriteLine("generated key length: " + aesKey.Length + " data: " + aesKeyBase64);
	}

  static byte[] GenerateRandomAesKey() {
    RNGCryptoServiceProvider rngCsp = new RNGCryptoServiceProvider();
    byte[] key = new byte[32];
    rngCsp.GetBytes(key);
    return key;
  }

  static string Base64Encoding(byte[] input) {
    return Convert.ToBase64String(input);
  }
}

